plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/view/repository/RepositoryList.js

```diff
@@ -25,14 +25,21 @@ Ext.define('NX.coreui.view.repository.RepositoryList', {
   ],
 
   store: 'NX.coreui.store.Repository',
-
+  
   columns: [
     {
       xtype: 'nx-iconcolumn',
       width: 36,
       iconVariant: 'x16',
-      iconName: function() {
-        return 'repository-default';
+      iconName: function(value, meta, record) {
+        switch((record.get('type'))) {
```

Comment:
This can be done simpler by not overriding iconName and setting:
iconNamePrefix: 'repository-'
dataIndex: 'type'